internal/server/run.go

```diff
@@ -0,0 +1,93 @@
+package server
+
+import (
+	"errors"
+	"io"
+	"log"
+	"net"
+	"tcp-load-balancer/internal/upstream"
+)
+
+var ErrUninitialized = errors.New("load balancer not initialized")
+
+// Run handles incoming connections until terminated.
+func (l *LoadBalancer) Run() error {
+	if l == nil || l.listener == nil {
+		return ErrUninitialized
+	}
+
+	for {
+		// TODO: Set up mTLS in next PR. For now, connect without TLS.
+		clientConn, err := l.listener.Accept()
+		if err != nil {
+			go respondAndClose(clientConn, err.Error())
+			continue
+		}
+
+		if err := l.handleConnection(clientConn); err != nil {
+			log.Printf("Unable to handle connection: %s", err)
+			go respondAndClose(clientConn, err.Error())
+		}
+	}
+}
+
+// handleConnection selects an upstream host, tracks connection counts, and forwards data upstream.
+func (l *LoadBalancer) handleConnection(clientConn net.Conn) error {
+	// Host selection is not included in goroutine handling so that requests arriving at the same time are not routed to the same host.
+	// This adds a small amount of latency to the request, but ensures accurate load balancing.
+	host, err := l.LeastConnections()
+	if err != nil {
+		return err
+	}
+
+	// Increment the connection count for the selected host.
+	if err = host.IncrementActiveConnections(); err != nil {
+		return err
+	}
+
+	// Copy data to the selected host, and decrement the connection count when the copy finishes.
+	go func() {
+		if err = forwardToHost(clientConn, host); err != nil {
```

Comment:
It looks like that right now the data are only copied from client to server and forwarding server data to the client is missing.

Reply:
Yes. This comment was included in the initial commit (line 55):
```
		// TODO: Support response data from hosts back to clients (outside scope of this project).
```

I didn't see anything in the project requirements that would indicate we needed to responding to clients, so I was shaving off a few lines of code, but it makes sense that would be a normal pattern. 

Should I include forwarding server data back to the client as part of project scope?

Reply:
Sounds good, I'll get that added.

Reply:
This has been added and tested now.